### PR TITLE
Remove useGithubStorage from GHES version checker method

### PR DESCRIPTION
**Pull request by begonaguereca**
Closes https://github.ghe.com/github/octoshift/issues/9649!

This pull request simplifies the method calls to AreBlobCredentialsRequired by removing the boolean parameter that checks for gitHubOwnedStorage.

While this produces the correct result, it is confusing for a reader. This PR refactors out that logic and adds the checks in the correct spots.

- [ ] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/gei/Commands/GenerateScript/GenerateScriptCommandHandler.cs [hunk 1]
```diff
@@ -93,14 +93,15 @@ private async Task<string> GenerateSequentialGithubScript(IEnumerable<(string Na
...
+
+        if (!useGithubStorage && await _ghesVersionChecker.AreBlobCredentialsRequired(ghesApiUrl))
         {
             if (awsBucketName.HasValue() || awsRegion.HasValue())
             {
                 content.AppendLine(VALIDATE_AWS_ACCESS_KEY_ID);
                 content.AppendLine(VALIDATE_AWS_SECRET_ACCESS_KEY);
             }
-            else
+            else if (!useGithubStorage)
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateScript/GenerateScriptCommandHandler.cs:104]: This check is now redundant 

```suggestion
```

File: src/gei/Commands/GenerateScript/GenerateScriptCommandHandler.cs [hunk 2]
```diff
@@ -131,14 +132,14 @@ private async Task<string> GenerateParallelGithubScript(IEnumerable<(string Name
...
-        if (await _ghesVersionChecker.AreBlobCredentialsRequired(ghesApiUrl))
+        if (!useGithubStorage && await _ghesVersionChecker.AreBlobCredentialsRequired(ghesApiUrl))
         {
             if (awsBucketName.HasValue() || awsRegion.HasValue())
             {
                 content.AppendLine(VALIDATE_AWS_ACCESS_KEY_ID);
                 content.AppendLine(VALIDATE_AWS_SECRET_ACCESS_KEY);
             }
-            else
+            else if (!useGithubStorage)
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateScript/GenerateScriptCommandHandler.cs:142]: same

```suggestion
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ public async Task Handle(MigrateRepoCommandArgs args)
 
         _log.LogInformation("Migrating Repo...");
 
-        var blobCredentialsRequired = await _ghesVersionChecker.AreBlobCredentialsRequired(args.GhesApiUrl, args.UseGithubStorage);
+        var blobCredentialsRequired = !args.UseGithubStorage || (await _ghesVersionChecker.AreBlobCredentialsRequired(args.GhesApiUrl));
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:67]: As described [here](https://github.com/github/gh-gei/pull/1293#discussion_r1838610636) we do need to always check the GHES version to be able to log the warnings.

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs [hunk 4]
```diff
@@ -396,7 +396,7 @@ private void ValidateGHESOptions(MigrateRepoCommandArgs args, bool cloudCredenti
 
             if (args.UseGithubStorage)
             {
-                _log.LogWarning("Ignoring --use-github-storage flag because you are running GitHub Enterprise Server (GHES) 3.8.0 or later. The blob storage credentials configured in your GHES Management Console will be used instead.");
+                _log.LogWarning("Provding the --use-github-storage flag will supersede any credentials you have configured in your GitHub Enterprise Server (GHES) Management Console.");
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:399]: as described [here](https://github.com/github/gh-gei/pull/1293#discussion_r1838631595) please test this logic to confirm that it's the case.

**Review by ArinGhazarian — CHANGES_REQUESTED** [review]

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs (hunk 3 above)
Comment (ArinGhazarian) [reply src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:67]: so this is what I mean:

```suggestion
        var blobCredentialsRequired = await _ghesVersionChecker.AreBlobCredentialsRequired(args.GhesApiUrl);
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs (hunk 4 above)
Comment (begonaguereca) [reply src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:399]: Manually tested and it is the case

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs [hunk 5]
```diff
@@ -116,7 +116,7 @@ public async Task Handle(MigrateRepoCommandArgs args)
               args.UseGithubStorage
             );
 
-            if (blobCredentialsRequired)
+            if (args.UseGithubStorage && blobCredentialsRequired)
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:119]: We need to display this message either if  we use the GH owned storage OR blob creds (`blobCredentialsRequired` is `true`):

```suggestion
            if (args.UseGithubStorage || blobCredentialsRequired)
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs [hunk 6]
```diff
@@ -221,7 +221,7 @@ private string ExtractGhesBaseUrl(string ghesApiUrl)
         var (gitArchiveUrl, metadataArchiveUrl, gitArchiveId, metadataArchiveId) = await _retryPolicy.Retry(
             async () => await GenerateArchive(githubSourceOrg, sourceRepo, skipReleases, lockSourceRepo));
 
-        if (!blobCredentialsRequired)
+        if (!useGithubStorage && !blobCredentialsRequired)
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:224]: We need a unit test for a scenario where `useGithubStorage` is `true` and `blobCredentialsRequired` is `false`

**Review by ArinGhazarian — COMMENTED** [review]
So once the two comments are addressed it should be all good to go the only missing thing is that I forgot to mention is the INT tests for `--use-github-storage` are only testing a single part upload so it'd be ideal if there is one additional test for testing a multipart upload. Moving forward here are our options:
1. Don’t add a test but run a manual functional test with a large archive > 1gb to make sure it 100% works (not recommended)
2. Add a new INT test that uses a pre-existing archive > 100MiB (this can be a new INT test not the Basic one that doesn’t download the archive and uses an existing one and we just need to test `--use-github-storage` scenario) (recommended)
3. if creating a > 100 MiB archive is not easily possible we can add a new test but change the archvie limit from 100 to say 10 MiB just to test the multipart upload.

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 7]
```diff
@@ -1632,5 +1632,107 @@ public async Task Sets_Target_Repo_Visibility()
...
+
+            _mockTargetGithubApi
+                .Setup(x => x.UploadArchiveToGithubStorage(githubOrgDatabaseId, It.Is<string>(a => a.EndsWith("git_archive.tar.gz")), gitContentStream).Result)
+                .Returns(uploadedGitArchiveUrl);
+
+            _mockTargetGithubApi
+                .Setup(x => x.UploadArchiveToGithubStorage(githubOrgDatabaseId, It.Is<string>(a => a.EndsWith("metadata_archive.tar.gz")), metaContentStream).Result)
+                .Returns(uploadedMetadataArchiveUrl);
+
+            Console.WriteLine($"MigrationSourceId: {migrationSourceId}, MigrationId: {migrationId}");
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:1715]: nit: probably lefover code as the `Console.WriteLine` won't work with `xUnit`

```suggestion
```

**Review by ArinGhazarian — APPROVED** [review]
Looks good just dropped a non-blocking nit.

**Review by ArinGhazarian — APPROVED** [review]

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs (hunk 4 above)
Comment (dylan-smith) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:399]:
```suggestion
                _log.LogWarning("Providing the --use-github-storage flag will supersede any credentials you have configured in your GitHub Enterprise Server (GHES) Management Console.");
```

**Review by ArinGhazarian — APPROVED** [review]

**Review by brianaj — APPROVED** [review]

**Pull request merged**